metadata/src/main/java/org/apache/kafka/metadata/util/SnapshotFileReader.java

```diff
@@ -130,17 +131,15 @@ private void handleControlBatch(FileChannelRecordBatch batch) {
             try {
                 short typeId = ControlRecordType.parseTypeId(record.key());
                 ControlRecordType type = ControlRecordType.fromTypeId(typeId);
-                switch (type) {
-                    case LEADER_CHANGE:
-                        LeaderChangeMessage message = new LeaderChangeMessage();
-                        message.read(new ByteBufferAccessor(record.value()), (short) 0);
-                        listener.handleLeaderChange(new LeaderAndEpoch(
+                if (Objects.requireNonNull(type) == ControlRecordType.LEADER_CHANGE) {
```

Comment:
Since ControlRecordType.fromTypeId always returns a non-null, we can just do `if(type == ControlRecordType.LEADER_CHANGE)`. Theoretically, using `Objects.requireNonNull(type)` changes the behavior by throwing instead of just logging an error.